test/unit/workspace/datastore_test.py

```diff
@@ -302,7 +300,7 @@ def test_get_resource_key(self):
     def test_get_resource_key_for_unknown_dataset_fails(self):
         """When get_resource_key() is called for unknown dataset it throws KeyError."""
         self.assertRaises(
-            KeyError, self.fetcher.get_resource_key, "unknown", "blob.zip"
+            AttributeError, self.fetcher.get_resource_key, "unknown", "blob.zip"
```

Comment:
I *think* we usually use the `pytest` specific code when catching errors in test:

```
with pytest.raises(AttributeError, match=r"some regex"):
    self.fetcher.get_resource_key("unknown", "blob.zip")
```

Though, as noted above, we seem to only use `get_resource_key()` in test, so maybe we should just privatize it and make sure we test its observed behavior through the public interface.

Reply:
We've tried to standardize on using `pytest` across the board but @rousik created these test classes originally using `unittest` and I was hoping not to refactor more than necessary to get this minor ergonomic improvement in.

Given that this method is literally used... nowhere but the tests, is there any reason not to just delete it entirely? Seems odd.

Reply:
Let's trash it!

Reply:
I've removed `get_resource_key()` and the related tests, and added another test which tries to get a DOI for a dataset that doesn't exist to verify that it fails as expected.